src/App/App.js

```diff
@@ -9,34 +11,72 @@ class App extends Component{
     super()
 
     this.state = {
-      userData: {},
-      displayLoginForm : false
+      userData:  {
+        id: null,
+        name: '',
+        email: ''
+      },
     }
   }
 
   buttonHandling = (event) => {
     if(event.target.innerHTML === 'Log in') {
-      this.setState({ displayLoginForm : true })
+      console.log('button')
     } 
 }
 
   authenticateUser = async (credentials) => {
     const promise = await fetcher.fetchUser(credentials)
-      // console.log('this is the promise', promise)
-     this.setState({ userData: promise, displayLoginForm : false})
-    //  console.log(this.state.userData.user.name)//this is getting the right info
+      this.setState({ userData: promise.user })
+      this.showHomepage()
+      // if (this.state.userData !== {}) {
+      // //  this.props.history.push('/')
+      // }
   }
 
+  showHomepage = () => {
+    console.log(this.props.history)
+  }
+
+  logOut = () => {
+    this.setState ( {
+      userData:  {
+        id: null,
+        name: '',
+        email: ''
+      }
+    })
+  }
+  
+
   render(){
-    const { displayLoginForm } = this.state
+    const { userData } = this.state;
+  
     return (
-     <>
-      <h1 className="login-info">User is <b>{displayLoginForm ? 'currently' : 'not'}</b> logged in.</h1>
-      <Homepage logIn={this.buttonHandling} name={this.state.userData}/> 
-      {displayLoginForm  && <LoginForm authenticateUser={this.authenticateUser}/> }
-    </>
-    );
+      <Router>
+        <h1 className="login-info"><b>{userData.name ? 'currently' : 'not'}</b> logged in</h1>
+        <Route 
```

Comment:
Route is the `alias` for `BrowserRouter`, check the line from above  when importing the `react-router`